pod/perlguts.pod

```diff
@@ -2717,6 +2717,24 @@ whatever the compiler has.
 If you are printing addresses of pointers, use UVxf combined
 with PTR2UV(), do not use %lx or %p.
 
+=head2 Formatted Printing of SVs
+
+If you want to print perl scalar which you get as C<SV*>, you can use
```

Comment:
This should probably have X<SVf>X<SVfARG> if there's not going to be any documentation for the individual macros (which would also be good)

Reply:
I do not know what you want to write more... I bring this attempt for documentation per @khwilliamson request from https://github.com/Dual-Life/Devel-PPPort/pull/151

Reply:
This made me realize that the pod I wrote for that is wrong.  UTF8f is really about strings, and SVf is more appropriate for SVs.  So I pushed 9bec17d7c14c525018251a18ce767f52c59cb84c to fix that.

Reply:
And I did some experiments with using SVf.  I came up with this more detailed patch.  How does it look?

`From 3baf5bfd0b6e14eab25491a49d71f165dcabb519 Mon Sep 17 00:00:00 2001
From: Karl Williamson <khw@cpan.org>
Date: Thu, 7 Nov 2019 13:54:48 -0700
Subject: [PATCH] Proposed pod for SVf format

---
 pod/perlguts.pod | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/pod/perlguts.pod b/pod/perlguts.pod
index 43aaff5593..2b194e3653 100644
--- a/pod/perlguts.pod
+++ b/pod/perlguts.pod
@@ -2717,6 +2717,28 @@ whatever the compiler has.
 If you are printing addresses of pointers, use UVxf combined
 with PTR2UV(), do not use %lx or %p.
 
+=head2 Formatted Printing of SVs
+
+The contents of SVs may be printed using the C<SVf> format, like so:
+
+ Perl_croak(aTHX_ "This croaked because: %" SVf "\n", SvfARG(err_msg))
+
+where C<err_msg> is an SV.
+
+Not all scalar types are printable.  Simple values certainly are: one of
+IV, UV, NV, or PV.  Also, if the SV is a pointer to some value,
+information about the type of that value and its address are displayed.
+The results of printing any other type of SV are undefined and likely to
+lead to an interpreter crash.  NVs are printed using a C<%g"-ish format.
+
+Note that the spaces are required around the C<SVf> in case the code is
+compiled with C++, to maintain compliance with its standard.
+
+Note that any filehandle being printed to under UTF-8 must be expecting
+UTF-8 in order to get good results and avoid Wide-character warnings.
+One way to do this for typical filehandles is to invoke perl with the
+C<-C>> parameter.  (See L<perlrun/-C [number/list]>.
+
 =head2 Formatted Printing of Strings
 
 If you just want the bytes printed in a NUL-terminated string, you can
-- 
2.17.1

`

Reply:
> And I did some experiments with using SVf. I came up with this more detailed patch. How does it look?

I would suggest to add example of concatenation of two scalars.

> if the SV is a pointer to some value, information about the type of that value and its address are displayed.

Instead `pointer` use `reference`.

And above is not truth for object references with overloaded stringify operator. For them %SVf use value from that operator instead of memory address.